Add Navigation menu to known entities

getdave opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
This PR adds a new `navigationMenus` entity (`wp_navigation`) to the list of ["known" entities](https://github.com/WordPress/gutenberg/blob/70f6f1924509b5a2389ac48d13e3eb00fae9a744/packages/core-data/src/entities.js#L21). This enables use of selector "shortcuts" to fetch Navigation Menus.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->


Spun out of https://github.com/WordPress/gutenberg/pull/39290/ in order to simplify that PR.

We often want to fetch data about Navigation Menus. To do this we have to write out the full `getEntityRecords()` providing all the correct parameters. This is rather verbose and requires remember all the correct params:

```js
wp.data.select('core').getEntityRecords('postType','wp_navigation')
```

Given that `wp_navigation` is now part of WP Core it makes sense to add a shortcut to enable shorthand selectors.

With this PR we can now achieve the same result with the following selector:

```js
wp.data.select('core').getNavigationMenus()
```

## How?

By adding the new entity to the root entity records `core-data` will generate a selector which will fetch Navigation Menus.

## Testing Instructions

First verify that Navigatoin Menus are available in the root entity records.

- Open dev tools console
- Type:
```
wp.data.select('core').getEntitiesByKind('root').map( ({kind,name}) => (name));
```
- Check for the `navigationMenu` entry.

Now check you can call the shortcut selector to return available Navigation posts.

- Create some Navigation Menus using the Nav block.
- Open devtools console
- Type:
```js
wp.data.select('core').getNavigationMenus()
```
- You should see all the available Navigation Menus (note you may have to run the selector twice in order to allow for the resolver to run).

Also verify you can utilise the singular form of the selector:

```
wp.data.select('core').getNavigationMenu(%%VALID_NAV_MENU_ID_HERE%%)
```



## Screenshots or screencast <!-- if applicable -->

getdave committed:
Add Navigation menu to known entities

adamziel commented:
We have menus, menu items, menu locations, navigation menus – it is getting confusing. Would you please add a few comments inside entities.js to explain how are these different from each other?

getdave committed:
Add comments for clarity

Addresses https://github.com/WordPress/gutenberg/pull/39498#issuecomment-1069104704

adamziel commented on packages/core-data/src/entities.js:
```diff
@@ -175,6 +175,14 @@ export const rootEntitiesConfig = [
 		key: 'plugin',
 	},
+	{
+		label: __( 'Navigation' ), // Block based Navigation Menus
```
Is it a post type? If so, do we refer to it with `kind="postType"` in other packages?

getdave replied:
It's a custom post type yes. That's why it's 

```
wp.data.select('core').getEntityRecords('postType','wp_navigation')
```

getdave replied:
It's a custom post type yes. That's why it's 

```
wp.data.select('core').getEntityRecords('postType','wp_navigation')
```

adamziel replied:
In this case this needs a follow up to replace all the `getEntityRecords('postType','wp_navigation')` calls for the `getEntityRecords('root','navigationMenus')`. 

But thinking more about this –  introducing two different ways to refer to the same thing will likely lead to, well, two different ways of using it. Shall we rollback this PR and instead hardcode a shortcut that would refer to the `postType, wp_navigation` pair?

getdave commented:
Rebasing in the hope that the e2e tests have been fixed on `trunk`.

getdave commented:
I revert this PR in https://github.com/WordPress/gutenberg/pull/39572 because it appeared to cause a regression whereby Templates wouldn't load in the Site Editor.
